tokio/Cargo.toml

```diff
@@ -92,6 +92,7 @@ time = []
 tokio-macros = { version = "~2.2.0", path = "../tokio-macros", optional = true }
 
 pin-project-lite = "0.2.11"
+rustversion = { git = "https://github.com/wenym1/rustversion", rev = "0b11410" }
```

Comment:
We are not going to add new dependencies. But let's talk about implementation once we have a test that clearly displays what the problem is.

Reply:
In this PR, the [Arc::into_inner](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.into_inner) is used, which is supported only after `stable 1.70.0`. I had to skip the newly added code for rust version before it. The dependency is added to conditionally skip the newly added code. If there is other way to achieve the same feature without adding the dependency it will be great.

The reason for using a patched version is that the latest release of the crate does not work on windows, which will can be fixed by a PR https://github.com/dtolnay/rustversion/pull/50 that uses the patched code.

Besides, `Arc::into_inner` is not supported in loom yet. To add the loom test, I had to implement `into_inner` support in a separate PR https://github.com/tokio-rs/loom/pull/354, which is also a patch to the loom crate used currently.